drift/core/beamtransfer.py

```diff
@@ -831,190 +831,260 @@ def _generate_svdfiles(self, regen=False, skip_svd_inv=False):
         ## m, performing the SVD, combining the beams and then write out the
         ## results.
 
-        # For each `m` collect all the `m` sections from each frequency file,
-        # and write them into a new `m` file. Use MPI if available.
-        for mi in mpiutil.mpirange(self.telescope.mmax + 1):
+        m_list = np.arange(self.telescope.mmax + 1)
+        if mpiutil.rank0:
+            # For each m, check whether the file exists, if so, whether we
+            # can open it. If these tests all pass, we can skip the file.
+            # Otherwise, we need to generate a new SVD file for that m.
+            for mi in m_list:
+                if os.path.exists(self._svdfile(mi)) and not regen:
+
+                    # File may exist but be un-openable, so we catch such an
+                    # exception. This shouldn't happen if we use caput.misc.lock_file(),
+                    # but we catch it just in case.
+                    try:
+                        fs = h5py.File(self._svdfile(mi), "r")
+                        fs.close()
+
+                        print(
+                            "m index %i. Complete file: %s exists. Skipping..."
+                            % (mi, (self._svdfile(mi)))
+                        )
+                        m_list[mi] = -1
+                    except:
+                        print(
+                            "m index %i. ***INCOMPLETE file: %s exists. Will regenerate..."
+                            % (mi, (self._svdfile(mi)))
+                        )
 
-            if os.path.exists(self._svdfile(mi)) and not regen:
-                print(
-                    "m index %i. File: %s exists. Skipping..."
-                    % (mi, (self._svdfile(mi)))
-                )
-                continue
-            else:
-                print("m index %i. Creating SVD file: %s" % (mi, self._svdfile(mi)))
+            # Reduce m_list to the m's that we need to compute
+            m_list = m_list[m_list != -1]
 
-            # Open m beams for reading.
-            fm = h5py.File(self._mfile(mi), "r")
+        # Broadcast reduced list to all tasks
+        m_list = mpiutil.bcast(m_list)
```

Comment:
same comment here. if had checked the files in parallel you don't need to bcast.